### PR TITLE
feat(Modelarts): add a new datasource to get models

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
Running tool: /usr/local/go/bin/go test -timeout 9000000000s -run ^TestAccDatasourceModels_basic$ github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts -v

=== RUN   TestAccDatasourceModels_basic
=== PAUSE TestAccDatasourceModels_basic
=== CONT  TestAccDatasourceModels_basic
--- PASS: TestAccDatasourceModels_basic (48.35s)
PASS
ok  	github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/modelarts	48.405s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/modelarts/data_source_huaweicloud_modelarts_models_test.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+}
+
+output "default_filter_is_useful" {
+  value = length(data.huaweicloud_modelarts_models.test.models) > 2
+}
+
+data "huaweicloud_modelarts_models" "name_filter" {
+  name        = huaweicloud_modelarts_model.test1.name
+  exact_match = "true"
+}
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/modelarts/data_source_huaweicloud_modelarts_models_test.go]: missing `depends_on` definition, this data source will execute ahead of the resource.
and the object of the alltrue function will be an empty slice, even if the result shows true

File: huaweicloud/services/acceptance/modelarts/data_source_huaweicloud_modelarts_models_test.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+data "huaweicloud_modelarts_models" "version_filter" {
+  version = huaweicloud_modelarts_model.test1.version
+}
+output "version_filter_is_useful" {
+  value = alltrue([for v in data.huaweicloud_modelarts_models.version_filter.models[*].version : v == "0.0.1"])
+}
+
+data "huaweicloud_modelarts_models" "status_filter" {
+  status = "published"
+}
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/modelarts/data_source_huaweicloud_modelarts_models_test.go]: ditto, and check all parameters (without `know after apply` behavior: Computed: true) of the data sources.

File: huaweicloud/services/acceptance/modelarts/data_source_huaweicloud_modelarts_models_test.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/modelarts/data_source_huaweicloud_modelarts_models_test.go]: ok, add depends_on

File: huaweicloud/services/acceptance/modelarts/data_source_huaweicloud_modelarts_models_test.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/acceptance/modelarts/data_source_huaweicloud_modelarts_models_test.go]: have checked all parameters in the case: `name_filter`

**Pull request merged**